File: Questions/versionControl.java
import java.util.*;

public class versionControl {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] logs = scanner.nextLine().split(",");

        Map<String, List<String>> hashmap = new HashMap<>();
        String currbranch = "";

        for (String log : logs) {
            int index = log.indexOf(" ");
            String cmd = log.substring(0, index).trim();
            String val = log.substring(index + 1).trim();

            if (cmd.equals("switch")) {
                if (!hashmap.containsKey(val)) {
                    hashmap.put(val, new ArrayList<>());
                }
                currbranch = val;
            } else if (cmd.equals("push")) {
                if (!hashmap.get(currbranch).contains(val)) {
                    hashmap.get(currbranch).add(val);
                }
            }
        }

        int maxi = 0;
        String res = "";
        for (String i : hashmap.keySet()) {
            int l = hashmap.get(i).size();
            if (l > maxi) {
                maxi = l;
                res = i;
            }
        }

        System.out.println(res);
    }
}

File: Questions/Python/versionControl.py
logs = list(map(str, input().split(',')))

hashmap = {}

currbranch = ""

for i in logs:
    index = i.index(" ")
    cmd = i[:index].strip()
    val = i[index+1:].strip()

    if cmd == "switch":
        if val not in hashmap:
            hashmap[val] = []
        curr = val
    elif cmd == "push":
        if val not in hashmap[curr]:
            hashmap[curr].append(val)

maxi = 0
res = ""
for i in hashmap.keys():
    l = len(hashmap[i])
    if l > maxi:
        maxi = l
        res = i

print(res)
